qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py

```diff
@@ -499,22 +580,30 @@ def get_raw(self) -> Tuple[Tuple[float, ...], Tuple[float, ...]]:
             self.root_instrument.write(MessageBuilder().fmt(1, 1).message)
             with self.root_instrument.timeout.set_to(new_timeout):
                 raw_data = self.instrument.ask(MessageBuilder().xe().message)
-                parsed_data = fmt_response_base_parser(raw_data)
         finally:
             self.root_instrument.write(MessageBuilder().fmt(fmt_format,
                                                             fmt_mode).message)
 
-        self.param1 = _FMTResponse(
-            *[parsed_data[i][::3] for i in range(0, 4)])
-        self.param2 = _FMTResponse(
-            *[parsed_data[i][1::3] for i in range(0, 4)])
-        self.source_voltage = _FMTResponse(
-            *[parsed_data[i][2::3] for i in range(0, 4)])
+        parsed_data = fmt_response_base_parser(raw_data)
 
-        self.shapes = ((len(self.source_voltage.value),),) * 2
-        self.setpoints = ((self.source_voltage.value,),) * 2
+        n_items_per_data_point = 4  # == len(_FMTResponse(None, None, None, None))
+        n_all_data_channels = n_channels + 1  # because sourced voltage values are also returned
+
+        for n in range(n_channels):
+            single_channel_data = _FMTResponse(
+                *[parsed_data[i][n::n_all_data_channels] for i in range(0, n_items_per_data_point)])
```

Comment:
If breaking just after the ( I would also put ) on a new line to make it more clear where the call to _FMTResponse ends. Alternatively I would extract the list generator into a variable

Reply:
done